--- a/win/utils.py
+++ b/win/utils.py
@@ -224,7 +224,7 @@ def vc_cmd(parms, cmd, arch=None, succeed=0):
     if arch == "x64":
         arch = "amd64"
     with ModEnv('PATH', "%s;%s\\VC;%s\\VC\\Auxiliary\\Build;" % (os.environ['PATH'], parms['MSVC_DIR'], parms['MSVC_DIR'])):
-        call('vcvarsall.bat %s && %s' % (arch, cmd), shell=True, succeed=succeed)
+        call('set "VSCMD_START_DIR=%%CD%%" && set "PATH=%s\\tools\\vswhere;%s\\ovpn3\\core\\impulse;%%PATH%%" && vswhere_vcvarsall.bat %s && %s' % (os.environ['SYSTEM_DEFAULTWORKINGDIRECTORY'], os.environ['SYSTEM_DEFAULTWORKINGDIRECTORY'], arch, cmd), shell=True, succeed=succeed)
 
 def vc_parms(parms, cmd_dict):
     cmd_dict["dbg_rel_flags"] = "/Zi" if parms['DEBUG'] else "/O2"
